src/html5/parser.rs

```diff
@@ -630,23 +635,24 @@ impl<'chars> Html5Parser<'chars> {
     fn process_html_content(&mut self) {
         if self.ignore_lf {
             if let Token::Text(value) = &self.current_token {
-                if value.eq(&"\n".to_string()) {
-                    self.current_token = self.fetch_next_token();
+                if value.starts_with('\n') {
+                    // We don't need to skip 1 char, but we can skip 1 byte, as we just checked for \n
+                    self.current_token = Token::Text(value.chars().skip(1).collect::<String>());
                 }
             }
             self.ignore_lf = false;
         }
 
-        // // Break when we reach the end of the token chars
-        // if self.current_token.is_eof() {
-        //     self.parser_finished = true;
-        // }
-
         match self.insertion_mode {
             InsertionMode::Initial => {
                 let mut anything_else = false;
 
                 match &self.current_token.clone() {
+                    Token::Text(value) if self.current_token.is_mixed() => {
```

Comment:
IMO, current_token.is_mixed shoud move fetch_next_token, and insert_tokens_at_queue_start could not need.

Reply:
@CharlesChen0823 I'm not sure what you exactly mean. Do you mean that we should do this "is_mixed" thing inside the `fetch_next_token` and not directly everywhere in the code?

Reply:
@CharlesChen0823 for now, I want to merge this PR.. if it's ok with you, can you describe in a PR draft what your idea is to change this?